src/NuGet.Core/NuGet.Protocol/Converters/JsonExtensions.cs

```diff
@@ -113,6 +114,23 @@ public static T GetJObjectProperty<T>(this JObject jobject, string propertyName)
             return targetProperty != null ? targetProperty.FromJToken<T>() : default(T);
         }
 
+        /// <summary>
+        /// Extract the property directly from JObject.
+        /// </summary>
+        /// <param name="jobject">The JObject which is the source</param>
+        /// <param name="prefixOfPropertyName">Prefix of the property name.</param>
+        /// <returns>The first property whose name starts with the prefix, if found; otherwise, the default value</returns>
+        public static JProperty PropertyWithMatchingPrefix(this JObject jobject, string prefixOfPropertyName)
```

Comment:
Please don't declare this in a public API.

The NuGet.Core parts are used as libraries and we should be mindful when extending the surface area.